nexus/reconfigurator/execution/src/clickhouse.rs

```diff
@@ -160,6 +160,38 @@ pub(crate) async fn deploy_nodes(
         "Successfully deployed all clickhouse server and keeper configs"
     );
 
+    // We only need to initialise the database schema into one of the ClickHouse replica
+    // servers as they are all part of the same cluster.
+    let Some(first_server_config) = server_configs.first() else {
+        let e = concat!(
+            "Failed to initialise database schema on the replicated ClickHouse cluster:",
+            " no replica server configuration file found");
+        error!(opctx.log, "{e}",);
+        return Err(vec![anyhow!(e)]);
+    };
+
+    let admin_addr = SocketAddr::V6(SocketAddrV6::new(
+        first_server_config.settings.listen_addr,
+        CLICKHOUSE_ADMIN_PORT,
+        0,
+        0,
+    ));
+    let admin_url = format!("http://{admin_addr}");
+    let log = opctx.log.new(slog::o!("admin_url" => admin_url.clone()));
+    let client = ClickhouseSingleClient::new(&admin_url, log.clone());
```

Comment:
Should this be `ClickhouseServerClient` instead of `ClickhouseSingleClient`? Right now they happen to have compatible APIs for this one method (`.init_db`), but we probably don't want to depend on that.

Reply:
whoops! good catch. Thanks!